psl-core/src/main/java/org/linqs/psl/application/inference/online/OnlineServer.java

```diff
@@ -72,6 +83,21 @@ public void start() {
         serverThread.blockUntilReady();
     }
 
+    /**
+     * Create a temporary file in the default temporary file directory.
+     * The creation of the temporary file may be used as a signal that this server is
+     * ready to accept client connections and online actions.
+     */
+    private void createServerTempFile() {
```

Comment:
Similar comments for the temp file in CLITest.
This needs to be safe for a mult-tenant machine, so use a scoped temp dir (`org.linqs.psl.util.SystemUtils.getTempDir()`).

Also, convention is to use a `.lock` extension to indicate that a process is running.
(If we were using Java 9, we would include the JVM's PID in the contents of the file.)